nmap-service-probes

```diff
@@ -15473,3 +15473,22 @@ Probe TCP VersionRequest q|VERSION|
 ports 3310
 rarity 8
 match clam m|^ClamAV ([\w.]+)/(\w+)/(.+)$| p/ClamAV/ v/$1 ($2)/ i/AV definitions updated on:$3/
+
```

Comment:
Ran a scan with those two new service probes and `--version-all` here's the results:

```
nmap --version-all -sV -n -d2 -p 4035,6715 10.10.0.201

Starting Nmap 7.40SVN ( https://nmap.org ) at 2017-01-19 11:03 PST
Fetchfile found /usr/local/bin/../share/nmap/nmap-services
Fetchfile found /usr/local/bin/../share/nmap/nmap.xsl
The max # of sockets we are using is: 0
--------------- Timing report ---------------
  hostgroups: min 1, max 100000
  rtt-timeouts: init 1000, min 100, max 10000
  max-scan-delay: TCP 1000, UDP 1000, SCTP 1000
  parallelism: min 0, max 0
  max-retries: 10, host-timeout: 0
  min-rate: 0, max-rate: 0
---------------------------------------------
NSE: Using Lua 5.3.
Fetchfile found /usr/local/bin/../share/nmap/nse_main.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/lpeg-utility.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/stdnse.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/strict.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/script.db
NSE: Arguments from CLI:
Fetchfile found /usr/local/bin/../share/nmap/scripts/allseeingeye-info.nse
NSE: Script allseeingeye-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/comm.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/shortport.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/unittest.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/nsedebug.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/listop.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/bin.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/bit.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/amqp-info.nse
NSE: Script amqp-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/amqp.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/match.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/bacnet-info.nse
NSE: Script bacnet-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/unicode.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/cccam-version.nse
NSE: Script cccam-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/formulas.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/db2-das-info.nse
NSE: Script db2-das-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/docker-version.nse
NSE: Script docker-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/json.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/http.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/base64.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/sasl.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/smbauth.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/slaxml.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/url.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/drda-info.nse
NSE: Script drda-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/drda.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/enip-info.nse
NSE: Script enip-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/ipOps.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/fingerprint-strings.nse
NSE: Script fingerprint-strings.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/fox-info.nse
NSE: Script fox-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/freelancer-info.nse
NSE: Script freelancer-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/http-server-header.nse
NSE: Script http-server-header.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/iax2-version.nse
NSE: Script iax2-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/ike-version.nse
NSE: Script ike-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/ike.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/jdwp-version.nse
NSE: Script jdwp-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/maxdb-info.nse
NSE: Script maxdb-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/tab.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/strbuf.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/mcafee-epo-agent.nse
NSE: Script mcafee-epo-agent.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/mqtt-subscribe.nse
NSE: Script mqtt-subscribe.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/mqtt.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/murmur-version.nse
NSE: Script murmur-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/ndmp-version.nse
NSE: Script ndmp-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/ndmp.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/netbus-version.nse
NSE: Script netbus-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/omron-info.nse
NSE: Script omron-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/openlookup-info.nse
NSE: Script openlookup-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/oracle-tns-version.nse
NSE: Script oracle-tns-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/ovs-agent-version.nse
NSE: Script ovs-agent-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/pptp-version.nse
NSE: Script pptp-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/quake1-info.nse
NSE: Script quake1-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/quake3-info.nse
NSE: Script quake3-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/rfc868-time.nse
NSE: Script rfc868-time.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/datetime.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/rpc-grind.nse
NSE: Script rpc-grind.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/rpc.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/datafiles.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/rpcinfo.nse
NSE: Script rpcinfo.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/s7-info.nse
NSE: Script s7-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/skypev2-version.nse
NSE: Script skypev2-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/snmp-info.nse
NSE: Script snmp-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/snmp.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/asn1.lua
Fetchfile found /usr/local/bin/../share/nmap/nselib/creds.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/stun-version.nse
NSE: Script stun-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/stun.lua
Fetchfile found /usr/local/bin/../share/nmap/scripts/teamspeak2-version.nse
NSE: Script teamspeak2-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/ventrilo-info.nse
NSE: Script ventrilo-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/wdb-version.nse
NSE: Script wdb-version.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/weblogic-t3-info.nse
NSE: Script weblogic-t3-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/scripts/xmpp-info.nse
NSE: Script xmpp-info.nse was selected by category.
Fetchfile found /usr/local/bin/../share/nmap/nselib/xmpp.lua
NSE: Loaded 40 scripts for scanning.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/allseeingeye-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/amqp-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/bacnet-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/cccam-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/db2-das-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/docker-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/drda-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/enip-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/fingerprint-strings.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/fox-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/freelancer-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/http-server-header.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/iax2-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/ike-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/jdwp-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/maxdb-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/mcafee-epo-agent.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/mqtt-subscribe.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/murmur-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/ndmp-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/netbus-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/omron-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/openlookup-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/oracle-tns-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/ovs-agent-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/pptp-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/quake1-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/quake3-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/rfc868-time.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/rpc-grind.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/rpcinfo.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/s7-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/skypev2-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/snmp-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/stun-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/teamspeak2-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/ventrilo-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/wdb-version.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/weblogic-t3-info.nse'.
NSE: Loaded '/usr/local/bin/../share/nmap/scripts/xmpp-info.nse'.
Fetchfile found /usr/local/bin/../share/nmap/nmap-payloads
Initiating Ping Scan at 11:03
Scanning 10.10.0.201 [2 ports]
ultrascan_host_probe_update called for machine 10.10.0.201 state UNKNOWN -> HOST_UP (trynum 0 time: 6993)
Changing ping technique for 10.10.0.201 to connect to port 443
Changing global ping host to 10.10.0.201.
Completed Ping Scan at 11:03, 0.01s elapsed (1 total hosts)
Overall sending rates: 280.82 packets / s.
Initiating Connect Scan at 11:03
Scanning 10.10.0.201 [2 ports]
Discovered open port 6715/tcp on 10.10.0.201
Discovered open port 4035/tcp on 10.10.0.201
Changing global ping host to 10.10.0.201.
Completed Connect Scan at 11:03, 0.01s elapsed (2 total ports)
Overall sending rates: 336.64 packets / s.
Fetchfile found /usr/local/bin/../share/nmap/nmap-service-probes
Initiating Service scan at 11:03
Scanning 2 services on 10.10.0.201
NSOCK INFO [0.4300s] nsock_iod_new2(): nsock_iod_new (IOD #1)
Starting probes against new service: 10.10.0.201:4035 (tcp)
NSOCK INFO [0.4300s] nsock_connect_tcp(): TCP connection requested to 10.10.0.201:4035 (IOD #1) EID 8
NSOCK INFO [0.4300s] nsock_iod_new2(): nsock_iod_new (IOD #2)
Starting probes against new service: 10.10.0.201:6715 (tcp)
NSOCK INFO [0.4310s] nsock_connect_tcp(): TCP connection requested to 10.10.0.201:6715 (IOD #2) EID 16
NSOCK INFO [0.4330s] nsock_trace_handler_callback(): Callback: CONNECT SUCCESS for EID 8 [10.10.0.201:4035]
Service scan sending probe NULL to 10.10.0.201:4035 (tcp)
NSOCK INFO [0.4330s] nsock_read(): Read request from IOD #1 [10.10.0.201:4035] (timeout: 6000ms) EID 26
NSOCK INFO [0.4340s] nsock_trace_handler_callback(): Callback: CONNECT SUCCESS for EID 16 [10.10.0.201:6715]
Service scan sending probe NULL to 10.10.0.201:6715 (tcp)
NSOCK INFO [0.4340s] nsock_read(): Read request from IOD #2 [10.10.0.201:6715] (timeout: 6000ms) EID 34
NSOCK INFO [6.4400s] nsock_trace_handler_callback(): Callback: READ TIMEOUT for EID 26 [10.10.0.201:4035]
Service scan sending probe RSE to 10.10.0.201:4035 (tcp)
NSOCK INFO [6.4400s] nsock_write(): Write request for 16 bytes to IOD #1 EID 43 [10.10.0.201:4035]
NSOCK INFO [6.4400s] nsock_read(): Read request from IOD #1 [10.10.0.201:4035] (timeout: 5000ms) EID 50
NSOCK INFO [6.4400s] nsock_trace_handler_callback(): Callback: READ TIMEOUT for EID 34 [10.10.0.201:6715]
Service scan sending probe JMON to 10.10.0.201:6715 (tcp)
NSOCK INFO [6.4400s] nsock_write(): Write request for 14 bytes to IOD #2 EID 59 [10.10.0.201:6715]
NSOCK INFO [6.4400s] nsock_read(): Read request from IOD #2 [10.10.0.201:6715] (timeout: 5000ms) EID 66
NSOCK INFO [6.4400s] nsock_trace_handler_callback(): Callback: WRITE SUCCESS for EID 43 [10.10.0.201:4035]
NSOCK INFO [6.4400s] nsock_trace_handler_callback(): Callback: WRITE SUCCESS for EID 59 [10.10.0.201:6715]
NSOCK INFO [6.5340s] nsock_trace_handler_callback(): Callback: READ SUCCESS for EID 66 [10.10.0.201:6715] (23 bytes): ACKNOWLEDGE01v09 2.9.5.
Service scan match (Probe JMON matched with JMON line 15494): 10.10.0.201:6715 is jmon.  Version: |JMON for zOS (FMID HALG300)|||
NSOCK INFO [6.5340s] nsock_iod_delete(): nsock_iod_delete (IOD #2)
NSOCK INFO [6.5450s] nsock_trace_handler_callback(): Callback: READ SUCCESS for EID 50 [10.10.0.201:4035] (63 bytes): ......@.......z@......@.......z@................@......@M.z.z.]
Service scan match (Probe RSE matched with RSE line 15484): 10.10.0.201:4035 is rse.  Version: |IBM Explorer for zOS (FMID HALG300)|||
NSOCK INFO [6.5450s] nsock_iod_delete(): nsock_iod_delete (IOD #1)
Completed Service scan at 11:03, 6.11s elapsed (2 services on 1 host)
NSE: Script scanning 10.10.0.201.
NSE: Starting runlevel 1 (of 2) scan.
Initiating NSE at 11:03
Completed NSE at 11:03, 0.01s elapsed
NSE: Starting runlevel 2 (of 2) scan.
Initiating NSE at 11:03
Completed NSE at 11:03, 0.00s elapsed
Nmap scan report for 10.10.0.201
Host is up, received conn-refused (0.0066s latency).
Scanned at 2017-01-19 11:03:35 PST for 6s
PORT     STATE SERVICE REASON  VERSION
4035/tcp open  rse     syn-ack IBM Explorer for zOS (FMID HALG300)
6715/tcp open  jmon    syn-ack JMON for zOS (FMID HALG300)
Final times for host: srtt: 6567 rttvar: 4608  to: 100000

Read from /usr/local/bin/../share/nmap: nmap-payloads nmap-service-probes nmap-services.
Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
Nmap done: 1 IP address (1 host up) scanned in 6.57 seconds
```